### PR TITLE
perf: efficient rendering program corrections

### PR DESCRIPTION
**Pull request by DADSILVER**
Modified how the rendering program is managed so that it works as the rest.

### CODE REVIEW

**Review by PaablooCH — APPROVED** [review]
works fine!!

File: Source/Modules/ModuleProgram.h [hunk 1]
```diff
@@ -19,31 +20,32 @@ class ModuleProgram : public Module
 	~ModuleProgram() override;
 
 	bool Start() override;
-
 	bool CleanUp() override;
 
-	void CreateProgram(unsigned int vtxShader, unsigned int frgShader);
+	void UpdateProgram(std::string& vtxShaderFileName, std::string& frgShaderFileName, int programType,
```
Comment (gerard-queralt) [inline Source/Modules/ModuleProgram.h:25]: This parameters can probably be const references

File: Source/Modules/ModuleProgram.h [hunk 2]
```diff
@@ -19,31 +20,32 @@ class ModuleProgram : public Module
...
 {
-	return programs[static_cast<int>(type)].get();
-}
-
-inline const unsigned ModuleProgram::GetProgram() const
-{
-	return program;
+	if (programs.size() != 0) 
+	{
+		return programs[static_cast<int>(type)].get();
```
Comment (gerard-queralt) [inline Source/Modules/ModuleProgram.h:45]: Should probably check the recieved type is not PROGRAM_TYPE_SIZE

File: Source/DataModels/Program/Program.cpp [hunk 3]
```diff
@@ -29,14 +33,15 @@ Program::Program(unsigned vertexShader, unsigned fragmentShader) : id (glCreateP
...
+	CleanUp();
 }
 
 void Program::Activate()
 {
 	glUseProgram(id);
 }
 
-void Program::Deactivate()
+void Program::Desactivate()
```
Comment (gerard-queralt) [inline Source/DataModels/Program/Program.cpp:44]: "Desactivate" is not a word, should be "Deactivate". Sorry to be picky :P

**Review by gerard-queralt — CHANGES_REQUESTED** [review]
Commented a few small things, but otherwise great changes!

File: Source/DataModels/Program/Program.cpp (hunk 3 above)
Comment (DADSILVER) [reply Source/DataModels/Program/Program.cpp:44]: Oky

**Review by gerard-queralt — APPROVED** [review]
Thank you David!

**Review by PaablooCH — APPROVED** [review]

**Pull request merged**